packages/story-editor/src/components/library/panes/pageTemplates/pageTemplate.js

```diff
@@ -120,6 +151,49 @@ function PageTemplate(
     setIsHover(false);
   }, []);
 
+  const imageUrl = page.image?.url || pageDataUrl;
+  const shouldPostBlob =
+    hasUploadMediaAction && pageDataUrl && !page.image?.url;
+  useEffect(() => {
+    if (!shouldPostBlob) {
+      return;
+    }
+
+    (async () => {
+      try {
+        const blob = await fetchRemoteBlob(pageDataUrl);
+        const file = blobToFile(blob, `${page.templateId}.jpg`, 'image/jpeg');
+        const resource = await uploadFile(file, {
+          web_stories_media_source: 'page-template',
+        });
+
+        updatePageTemplate(page.templateId, {
+          featured_media: resource.id,
+        });
+      } catch (err) {
```

Comment:
🤔 just a question here for me

If this step fails, does this means the template never gets saved as an image?

Reply:
They'll get it saved as an image, but maybe just not in this users session if an error occurs. Basically once any user goes to check out their saved templates, their machine should attempt to convert all of them and upload them to wp-media. once it's uploaded to wp-media, every user will see the saved template as an image.

so if it errors, they won't _never_ get the template saved as an image, but the template just won't get saved to wp-media in that session